broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/TestPartitionTransitionContext.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import io.atomix.raft.RaftServer.Role;
+import io.atomix.raft.partition.RaftPartition;
+import io.camunda.zeebe.broker.PartitionListener;
+import io.camunda.zeebe.broker.exporter.stream.ExporterDirector;
+import io.camunda.zeebe.broker.system.partitions.PartitionContext;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionContext;
+import io.camunda.zeebe.broker.system.partitions.StateController;
+import io.camunda.zeebe.broker.system.partitions.impl.AsyncSnapshotDirector;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.engine.processing.streamprocessor.StreamProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecord;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessorFactory;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.CommandResponseWriter;
+import io.camunda.zeebe.logstreams.log.LogStream;
+import io.camunda.zeebe.snapshots.ConstructableSnapshotStore;
+import io.camunda.zeebe.util.health.HealthMonitor;
+import io.camunda.zeebe.util.sched.ActorSchedulingService;
+import io.camunda.zeebe.util.sched.TestConcurrencyControl;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class TestPartitionTransitionContext implements PartitionTransitionContext {
+
+  private RaftPartition raftPartition;
+  private Role currentRole;
+  private long currentTerm;
+  private HealthMonitor healthMonitor;
+  private TypedRecordProcessorFactory streamProcessorFactory;
+  private ExporterDirector exporterDirector;
+  private LogStream logStream;
+  private StreamProcessor streamProcessor;
+  private ActorSchedulingService actorSchedulingService;
+  private ZeebeDb zeebeDB;
+  private TestConcurrencyControl concurrencyControl;
+  private StateController stateController;
+
+  @Override
+  public int getPartitionId() {
+    return 1;
+  }
+
+  @Override
+  public RaftPartition getRaftPartition() {
+    return raftPartition;
+  }
+
+  @Override
+  public List<ActorFuture<Void>> notifyListenersOfBecomingLeader(final long newTerm) {
+    return null;
```

Comment:
```suggestion
    return TestActorFuture.completedFuture(null);
```
same for `...OfBecomingFollower`